jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/CachedJMSConnectionFactory.java

```diff
@@ -38,6 +37,37 @@
 public class CachedJMSConnectionFactory extends JMSConnectionFactory {
 
     /**
+     * Parameter for passing in cache level.
+     */
+    public static final String PARAM_CACHE_LEVEL = "transport.jms.CacheLevel";
+
+    /**
+     * Do not cache any JMS resources between tasks (when sending) or JMS CF's
+     * (when sending)
+     */
+    public static final int CACHE_NONE = 0;
+    /**
+     * Cache only the JMS connection between tasks (when receiving), or JMS CF's
+     * (when sending)
+     */
+    public static final int CACHE_CONNECTION = 1;
```

Comment:
These values are valid only within the CachedJMSConnectionFactory class. Adding an enum would not add any advantage in this case.